src/column.rs

```diff
@@ -278,18 +296,25 @@ impl Column {
 				}
 			};
 
-			if let Some((table, sub_index, existing_tier, existing_address)) = existing {
+			if let Some((table, sub_index, existing_tier, existing_address)) = existing_index {
 				let existing_tier = existing_tier as usize;
 				if self.collect_stats {
+					// [Review] seems rather costy for the unsized value. Could consider storing size for
+					// those multipart value.
 					let cur_size = tables.value[existing_tier].size(&key, existing_address.offset(), log)?.unwrap_or(0);
 					self.stats.replace_val(cur_size, val.len() as u32);
 				}
 				if self.ref_counted {
+					// [Review] no check value is equal: means ref_count only works when preimage is true?
+					// should we 'debug_assert!(target_tier == existing_tier);'?
```

Comment:
Instead I'd just document that inserting an existing ref-counted value will bump the counter and ignore the new value. Or maybe provide an additional `inc_ref` function to avoid using `insert`. Substrate wrapper around the database API already has similar semantics.